v2/api/documentdb/v1beta20210515/sql_role_assignment_types_gen.go

```diff
@@ -330,7 +343,12 @@ type DatabaseAccounts_SqlRoleAssignment_Spec struct {
 
 	// PrincipalId: The unique identifier for the associated AAD principal in the AAD graph to which access is being granted
 	// through this Role Assignment. Tenant ID for the principal is inferred using the tenant associated with the subscription.
-	PrincipalId *string `json:"principalId,omitempty"`
+	PrincipalId *string `json:"principalId,omitempty" optionalConfigMapPair:"PrincipalId"`
+
+	// PrincipalIdConfigRef: The unique identifier for the associated AAD principal in the AAD graph to which access is being
+	// granted through this Role Assignment. Tenant ID for the principal is inferred using the tenant associated with the
+	// subscription.
+	PrincipalIdConfigRef *genruntime.ConfigMapReference `json:"principalIdConfigRef,omitempty" optionalConfigMapPair:"PrincipalId"`
```

Comment:
I think now we can have [sqlRoleAssingmnet](https://github.com/Azure/azure-service-operator/blob/7316b37c56d2ba998f12bc3f096ae672dd847300/v2/internal/testcommon/samples_tester.go#L47) sample tested?

Reply:
Also, would be worth adding these config ref to the samples for authorization/RoleAssignment as well for the users to know how to use it.

Reply:
Yeah good point, users are basically always gonna wanna do this

Reply:
I will do this in a follow-up PR though, just to keep this PR clean. I also need to do some docs work for this stuff so will do that too in the same PR